api/src/core/v1.rs

```diff
@@ -13,34 +23,259 @@ pub const GROUP_VERSION: GroupVersion = GroupVersion {
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PodLogOptions {
+    #[serde(skip_serializing_if = "is_default")]
+    pub container: Option<String>,
+    #[serde(skip_serializing_if = "is_default")]
+    pub follow: bool,
+    #[serde(skip_serializing_if = "is_default")]
+    pub previous: bool,
+    #[serde(skip_serializing_if = "is_default")]
+    pub since_seconds: Option<i64>,
+    #[serde(skip_serializing_if = "is_default")]
+    pub since_time: Option<Time>,
+    #[serde(skip_serializing_if = "is_default")]
+    pub timestamps: bool,
+    #[serde(skip_serializing_if = "is_default")]
+    pub tail_lines: Option<i64>,
+    #[serde(skip_serializing_if = "is_default")]
+    pub limit_bytes: Option<i64>,
+}
+
+fn booltrue() -> bool {
+    true
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PodAttachOptions {
+    #[serde(skip_serializing_if = "is_default")]
+    pub stdin: bool,
+    #[serde(skip_serializing_if = "is_default", default = "booltrue")]
+    pub stdout: bool,
+    #[serde(skip_serializing_if = "is_default", default = "booltrue")]
+    pub stderr: bool,
+    #[serde(skip_serializing_if = "is_default", rename = "TTY")]
+    pub tty: bool,
+    #[serde(skip_serializing_if = "is_default")]
+    pub container: Option<String>,
 }
 
-pub trait Resource {
-    type List: List;
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PodExecOptions {
+    #[serde(skip_serializing_if = "is_default")]
+    pub stdin: bool,
+    #[serde(skip_serializing_if = "is_default", default = "booltrue")]
+    pub stdout: bool,
+    #[serde(skip_serializing_if = "is_default", default = "booltrue")]
+    pub stderr: bool,
+    #[serde(skip_serializing_if = "is_default", rename = "TTY")]
+    pub tty: bool,
+    #[serde(skip_serializing_if = "is_default")]
+    pub container: Option<String>,
+    #[serde(skip_serializing_if = "is_default")]
+    pub command: Vec<String>,
+}
 
-    fn gvr(&self) -> GroupVersionResource;
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PodPortForwardOptions {
+    #[serde(skip_serializing_if = "is_default")]
+    pub ports: Vec<i32>,
 }
 
-impl NamespacedResource for Pods {
-    type List = PodList;
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PodProxyOptions {
+    #[serde(skip_serializing_if = "is_default")]
+    pub path: String,
+}
 
-    fn namespaced(&self) -> bool {
-        true
+// TODO: The stream errors should probably be some other error type.
+// Also: StreamItem should probably be a Buf, or replace the whole
+// thing with AsyncRead/Write.
+pub type ByteStream = Box<Stream<Item = Vec<u8>, Error = Error> + Send>;
```

Comment:
It would be nice to show this as unstable someway

Reply:
I don't think we have a way to do this (I don't think std's `#[unstable]` annotation means anything in user crates) - apart from what the crate-level "0.1" semver communicates (there is no API stability at present).